src/waveform/renderers/waveformrendermark.cpp

```diff
@@ -41,7 +41,7 @@ void WaveformRenderMark::draw(QPainter* painter, QPaintEvent* /*event*/) {
     for (int i = 0; i < m_marks.size(); i++) {
         WaveformMarkPointer mark = m_marks[i];
 
-        if (!mark->m_pPointCos)
+        if (mark->is_m_pPointCos_Null())
```

Comment:
This can be checked inside getPlayPosition(); and return 0.0 or -1 .0
Why we cannot set a Mark at 0.0 by the way. IMHO it is a valid use-case to have a cue point at 0.0 or even in the preroll.